### PR TITLE
soc/intel_adsp: cavstool: add support for DSP power state: D3

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits, last one:

Change some behaviors when --log-only is passed in order to add support
for DSP power state: D3. This makes it possible to keep `cavstool
--log-only` running permanently in the "real-world" use case where the
Linux kernel loads the firmware and powers down audio when not in use.

More specifically:
- Do not disable D3 when using --log-only.
- Wait forever for the FW at boot time.
- Check live status when an invalid IPC is received and wait forever if
  not alive.

### CODE REVIEW

File: soc/xtensa/intel_adsp/tools/cavstool.py [hunk 1]
```diff
@@ -648,7 +657,7 @@ async def _main(server):
     log.info(f"Detected cAVS {'1.5' if cavs15 else '1.8+'} hardware")
 
     if args.log_only:
-        wait_fw_entered()
+        wait_fw_entered(timeout_s=None)
```
Comment (andyross) [inline soc/xtensa/intel_adsp/tools/cavstool.py:660]: Do we really need to wait "forever"?  Surely something eventually has to time out and give up, right?  Or is the expectation that that happens externally in a CI rig or whatever?

**Review by andyross — DISMISSED** [review]
Makes sense.  It's nice to see the PCI layer PM in this script getting some attention, most of it was voodoo as I don't know that layer in linux well at all.

**Review by teburd — COMMENTED** [review]
The change itself is fine, I'd just like to see more details in the commit messages. I can't recall all the details of our conversation as to why this is useful and that was only a little bit ago.

e.g. What specific problem was there prior to this change. I imagine its really useful when running this as a logger only with sof firmware as the kernel may change the power state? Some details of that in the commit message would be much appreciated.

File: soc/xtensa/intel_adsp/tools/cavstool.py (hunk 1 above)
Comment (marc-hb) [reply soc/xtensa/intel_adsp/tools/cavstool.py:660]: Yes the idea is really to `--log-only` forever. Either in interactive use and then Ctrl-C is what's expected, or in a "CI rig" and then there will be plenty of other timeouts in various other layers, really no need to go and try to guess what a good default would be there. I mean no way to predict all the test scenarios this small cog will be a part of.

For instance https://github.com/thesofproject/sof-test is already used to leave sof-logger running forever in the background and kill it when it's done (and before starting a new test).

**Review by andyross — APPROVED** [review]
Still +1

**Review by teburd — APPROVED** [review]

**Pull request merged**